tools/pd-ctl/pdctl/command/component_command.go

```diff
@@ -112,15 +112,29 @@ func deleteComponentConfigCommandFunc(cmd *cobra.Command, args []string) {
 }
 
 func getComponentIDCommandFunc(cmd *cobra.Command, args []string) {
-	if len(args) != 1 {
+	argsLen := len(args)
+	// args too long
+	if argsLen > 1 {
 		cmd.Usage()
 		return
 	}
+	var (
+		hasIdsVal = false
+		ids       = "all"
+	)
+	if argsLen == 1 {
+		hasIdsVal = true
+		ids = args[0]
+	}
+	prefix := path.Join(componentConfigPrefix, "ids", ids)
 
-	prefix := path.Join(componentConfigPrefix, "ids", args[0])
 	r, err := doRequest(cmd, prefix, http.MethodGet)
 	if err != nil {
-		cmd.Printf("Failed to get component %s's id: %s\n", args[0], err)
+		if hasIdsVal {
```

Comment:
Ditto, we can directly use the length of args to do it.

Reply:
Is there anything else I can do for this pr? 📲

Reply:
This comment has not solved yet?

Reply:
Sorry, it has been updated.